offset_manager.go

```diff
@@ -329,6 +336,17 @@ func (pom *partitionOffsetManager) MarkOffset(offset int64, metadata string) {
 	}
 }
 
+func (pom *partitionOffsetManager) ResetOffset(offset int64, metadata string) {
+	pom.lock.Lock()
+	defer pom.lock.Unlock()
+
+	if offset < pom.offset {
```

Comment:
@eapache @fgeller sorry I'm late to the party on this, just now getting around to implement our usage of this feature.

Any reason for this condition? this is preventing me from updating the metadata without changing the offset.

Reply:
> updating the metadata without changing the offset.

I don't think either of us ever considered a use case for that. I suppose it would be easy enough to make this a `<=`

Reply:
https://github.com/Shopify/sarama/commit/96fa1c837819761fc5c706010a9677aab15ca46d